server/sonar-db-dao/src/main/java/org/sonar/db/MyBatis.java

```diff
@@ -0,0 +1,301 @@
+import org.sonar.db.organization.OrganizationMapper;
+import org.sonar.db.permission.AuthorizationMapper;
+import org.sonar.db.permission.GroupPermissionDto;
+import org.sonar.db.permission.GroupPermissionMapper;
+import org.sonar.db.permission.UserPermissionDto;
+import org.sonar.db.permission.UserPermissionMapper;
+import org.sonar.db.permission.template.PermissionTemplateCharacteristicDto;
+import org.sonar.db.permission.template.PermissionTemplateCharacteristicMapper;
+import org.sonar.db.permission.template.PermissionTemplateDto;
+import org.sonar.db.permission.template.PermissionTemplateGroupDto;
+import org.sonar.db.permission.template.PermissionTemplateMapper;
+import org.sonar.db.permission.template.PermissionTemplateUserDto;
+import org.sonar.db.property.InternalPropertiesMapper;
+import org.sonar.db.property.InternalPropertyDto;
+import org.sonar.db.property.PropertiesMapper;
+import org.sonar.db.property.ScrapPropertyDto;
+import org.sonar.db.purge.IdUuidPair;
+import org.sonar.db.purge.PurgeMapper;
+import org.sonar.db.purge.PurgeableAnalysisDto;
+import org.sonar.db.qualitygate.ProjectQgateAssociationDto;
+import org.sonar.db.qualitygate.ProjectQgateAssociationMapper;
+import org.sonar.db.qualitygate.QualityGateConditionDto;
+import org.sonar.db.qualitygate.QualityGateConditionMapper;
+import org.sonar.db.qualitygate.QualityGateDto;
+import org.sonar.db.qualitygate.QualityGateMapper;
+import org.sonar.db.qualityprofile.ActiveRuleDto;
+import org.sonar.db.qualityprofile.ActiveRuleMapper;
+import org.sonar.db.qualityprofile.ActiveRuleParamDto;
+import org.sonar.db.qualityprofile.QProfileChangeMapper;
+import org.sonar.db.qualityprofile.QualityProfileDto;
+import org.sonar.db.qualityprofile.QualityProfileMapper;
+import org.sonar.db.rule.RuleDto;
+import org.sonar.db.rule.RuleMapper;
+import org.sonar.db.rule.RuleParamDto;
+import org.sonar.db.rule.RuleRepositoryMapper;
+import org.sonar.db.schemamigration.SchemaMigrationDto;
+import org.sonar.db.schemamigration.SchemaMigrationMapper;
+import org.sonar.db.source.FileSourceMapper;
+import org.sonar.db.user.AuthorDto;
+import org.sonar.db.user.AuthorMapper;
+import org.sonar.db.user.GroupDto;
+import org.sonar.db.user.GroupMapper;
+import org.sonar.db.user.GroupMembershipDto;
+import org.sonar.db.user.GroupMembershipMapper;
+import org.sonar.db.user.RoleMapper;
+import org.sonar.db.user.UserDto;
+import org.sonar.db.user.UserGroupDto;
+import org.sonar.db.user.UserGroupMapper;
+import org.sonar.db.user.UserMapper;
+import org.sonar.db.user.UserTokenCount;
+import org.sonar.db.user.UserTokenDto;
+import org.sonar.db.user.UserTokenMapper;
+import org.sonar.db.webhook.WebhookDeliveryMapper;
+
+public class MyBatis {
+
+  private final Database database;
+  private SqlSessionFactory sessionFactory;
+
+  public MyBatis(Database database) {
+    this.database = database;
+  }
+
+  // FIXME should be visible only to DAOs -> to be moved to AbstractDao
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1134)